cmd/api/src/api/v2/auth/saml.go

```diff
@@ -396,6 +457,40 @@ func (s ManagementResource) SAMLCallbackHandler(response http.ResponseWriter, re
 				// SAML credentials issue scenario again
 				redirectToLoginPage(response, request, "Your SSO was unable to authenticate your user, please contact your Administrator")
 			} else {
+				if ssoProvider.Config.AutoProvision.Enabled {
+					if user, err := s.db.LookupUser(request.Context(), principalName); err != nil {
+						if errors.Is(err, database.ErrNotFound) {
```

Comment:
Check errors.Is first, then check err nil after, but not inside. Again, this led to an unchecked error case